src/test/java/test/Test1.java

```diff
@@ -36,7 +38,8 @@ public void includedGroups() {
 
         tng.run();
 
-        assertThat(listener.getSucceedMethodNames()).containsExactly("method1", "method3");
+        List<String> expected = Arrays.asList("method1", "method3");
+        doAssert(listener.getSucceedMethodNames(), expected);
```

Comment:
@juherr - As per [this](https://stackoverflow.com/a/1097826/679824) StackOverFlow post, it looks like the order is not defined. Since the order is not defined, there's no predicting on how it could be.

> Just an idea: It could be something like "alphaOrder" where methods of a class will be sorted by their name and linked together.

Sounds good. But I would be glad if you could please help call out as to where would this option/configuration reside. Is it from within the TestNG DTD or via a JVM argument or something on that lines ? The JVM argument would be the easiest to do.